client/blocks/plugins-scheduled-updates/hooks/use-create-monitor.tsx

```diff
@@ -18,33 +47,39 @@ export function useCreateMonitor( siteSlug: SiteSlug ) {
 	const { createMonitorSettings } = useCreateMonitorSettingsMutation( siteSlug );
 
 	const createMonitor = () => {
-		activateModule(
-			siteId,
-			'monitor',
-			true
-		)( ( args: { type: string } ) => {
-			if ( args.type === JETPACK_MODULE_ACTIVATE_SUCCESS ) {
-				setTimeout( () => {
-					createMonitorSettings( {
-						urls: [
-							{
-								// The home URL needs to be one of the URLs monitored.
-								check_interval: CRON_CHECK_INTERVAL,
-								monitor_url: siteUrl,
-							},
-							{
-								// Monitoring the wp-cron.php file to ensure that the cron jobs are running.
-								check_interval: CRON_CHECK_INTERVAL,
-								monitor_url: siteUrl + '/wp-cron.php',
-							},
-						],
-					} as UpdateMonitorSettings );
-				}, 3000 );
-			}
-		} );
+		activateMonitorModule( siteId!, () =>
+			createMonitorSettings( {
+				urls: createMonitorUrls( siteUrl! ),
+			} )
+		);
 	};
 
 	return {
 		createMonitor,
 	};
 }
+
+export function useCreateMonitors( siteSlugs: SiteSlug[] ) {
+	const siteIds = useSelector( ( state ) => siteSlugs.map( ( slug ) => getSiteId( state, slug ) ) );
+	const siteUrls = useSelector( ( state ) =>
+		siteIds.map( ( siteId ) => getSiteUrl( state, siteId as number ) )
+	);
+	const { createMonitorSettings } = useBatchCreateMonitorSettingsMutation( siteSlugs );
```

Comment:
Do we need to pass the `siteSlugs` as a params in `useBatchCreateMonitorSettingsMutation` function here?

Reply:
I was wrestling a lot with [the rules of hooks](https://legacy.reactjs.org/docs/hooks-rules.html) yesterday. I'm going to refactor this, I thought of a better way 😀

Reply:
You're right, not needed!